src/prefect/cli/deployment.py

```diff
@@ -267,36 +267,53 @@ async def set_schedule(
     """
     assert_deployment_name_format(name)
 
-    interval_schedule = {
-        "interval": interval,
-        "interval_anchor": interval_anchor,
-        "timezone": timezone,
-    }
-    cron_schedule = {"cron": cron_string, "day_or": cron_day_or, "timezone": timezone}
-    if rrule_string is not None:
-        rrule_schedule = json.loads(rrule_string)
-        if timezone:
-            # override timezone if specified via CLI argument
-            rrule_schedule.update({"timezone": timezone})
-    else:
-        # fall back to empty schedule dictionary
-        rrule_schedule = {"rrule": None}
+    if sum(option is not None for option in [interval, rrule_string, cron_string]) != 1:
+        exit_with_error(
+            "Exactly one of `--interval`, `--rrule`, or `--cron` must be provided."
+        )
 
-    def updated_schedule_check(schedule):
-        return any(v is not None for k, v in schedule.items() if k != "timezone")
+    if interval_anchor and not interval:
+        exit_with_error("An anchor date can only be provided with an interval schedule")
 
-    updated_schedules = list(
-        filter(
-            updated_schedule_check, (interval_schedule, cron_schedule, rrule_schedule)
+    if interval is not None:
+        if interval_anchor:
+            try:
+                pendulum.parse(interval_anchor)
+            except ValueError:
+                exit_with_error("The anchor date must be a valid date string.")
```

Comment:
Does this need to be updated now that `main` has these changes?

Reply:
Yes!